src/Client.js

```diff
@@ -43,36 +40,12 @@ export default class Client {
 
   /**
    * Add a provider
-   * @param {!string|Provider} provider - The provider instance or RPC connection string
+   * @param {!Provider} provider - The provider instance or RPC connection string
    * @return {Client} Returns instance of Client
    * @throws {InvalidProviderError} When invalid provider is provider
    * @throws {DuplicateProviderError} When same provider is added again
    */
   addProvider (provider) {
-    if (isString(provider)) {
```

Comment:
I'd like to keep this [proto]://[user]:[pass]@[host]:[port]

Reply:
Is there a particular reason? 
As well as having to test and support this functionality going forward, it's very specialised:
- It only works for RPC providers
- It only works for RPC providers provided internally by the library that follow the same argument order
- It does not support providing any additional arguments that an RPC provider might take

Reply:
I guess we can remove it, however, I hardly find our own libs useful. I actually ended up creating a very simple and slimmed down version for myself https://github.com/h4v0kr/multichain-rpc